Counterexamples/CliffordAlgebra_not_injective.lean

```diff
@@ -0,0 +1,273 @@
+import Mathlib.RingTheory.MvPolynomial.Basic
+import Mathlib.RingTheory.MvPolynomial.Ideal
+
+/-! # `algebraMap R (CliffordAlgebra Q)` is not always injective.
+
+A formalization of https://mathoverflow.net/questions/60596/clifford-pbw-theorem-for-quadratic-form/87958#87958
+
+Some Zulip discussion at https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/.F0.9D.94.BD.E2.82.82.5B.CE.B1.2C.20.CE.B2.2C.20.CE.B3.5D.20.2F.20.28.CE.B1.C2.B2.2C.20.CE.B2.C2.B2.2C.20.CE.B3.C2.B2.29/near/222716333.
+-/
+
+noncomputable section
+
+open scoped BigOperators
+
+namespace Q60596
+
+open MvPolynomial
+
+/-- The monomial ideal generated by terms of the form $x_ix_i$. -/
+def kIdeal : Ideal (MvPolynomial (Fin 3) (ZMod 2)) :=
+  Ideal.span (Set.range fun i => (X i * X i : MvPolynomial (Fin 3) (ZMod 2)))
+
+theorem mem_kIdeal_iff (x : MvPolynomial (Fin 3) (ZMod 2)) :
+    x ∈ kIdeal ↔ ∀ m : Fin 3 →₀ ℕ, m ∈ x.support → ∃ i, 2 ≤ m i := by
+  have :
+      kIdeal = Ideal.span ((monomial · (1 : ZMod 2)) '' Set.range (Finsupp.single · 2)) := by
+    simp_rw [kIdeal, X, monomial_mul, one_mul, ← Finsupp.single_add, ← Set.range_comp,
+      Function.comp]
+  rw [this, mem_ideal_span_monomial_image]
+  simp
+
+theorem X0_X1_X2_nmem_kIdeal : (X 0 * X 1 * X 2 : MvPolynomial (Fin 3) (ZMod 2)) ∉ kIdeal := by
+  intro h
+  simp_rw [mem_kIdeal_iff, support_mul_X, support_X, Finset.map_singleton, addRightEmbedding_apply,
+    Finset.mem_singleton, forall_eq, ← Fin.sum_univ_three fun i => Finsupp.single i 1,
+    ← Finsupp.equivFunOnFinite_symm_eq_sum] at h
+
+theorem mul_self_mem_kIdeal_of_X0_X1_X2_mul_mem {x : MvPolynomial (Fin 3) (ZMod 2)}
+    (h : X 0 * X 1 * X 2 * x ∈ kIdeal) : x * x ∈ kIdeal := by
+  rw [mem_kIdeal_iff] at h
+  have : x ∈ Ideal.span ((X : Fin 3 → MvPolynomial _ (ZMod 2)) '' Set.univ) := by
+    rw [mem_ideal_span_X_image]
+    intro m hm
+    simp_rw [mul_assoc, support_X_mul, Finset.map_map, Finset.mem_map,
+      Function.Embedding.trans_apply, addLeftEmbedding_apply, forall_exists_index,
+      and_imp, forall_apply_eq_imp_iff₂, ← add_assoc, ←
+      Fin.sum_univ_three fun i => Finsupp.single i 1, ← Finsupp.equivFunOnFinite_symm_eq_sum,
+      Finsupp.add_apply, Finsupp.equivFunOnFinite_symm_apply_toFun] at h
+    refine (h _ hm).imp fun i hi => ⟨Set.mem_univ _, ?_⟩
+    rintro hmi
+    rw [hmi] at hi
+    norm_num at hi
+  rw [as_sum x, CharTwo.sum_mul_self]
+  refine sum_mem fun m hm => ?_
+  rw [mem_kIdeal_iff, monomial_mul]
+  intro m' hm'
+  obtain rfl := Finset.mem_singleton.1 (support_monomial_subset hm')
+  rw [mem_ideal_span_X_image] at this
+  obtain ⟨i, _, hi⟩ := this m hm
+  simp_rw [←one_add_one_eq_two]
+  refine ⟨i, Nat.add_le_add ?_ ?_⟩ <;> rwa [Nat.one_le_iff_ne_zero]
+
+/-- `𝔽₂[α, β, γ] / (α², β², γ²)` -/
+def K : Type _ := _ ⧸ kIdeal
+
+instance : CommRing K := Ideal.Quotient.commRing _
+
+theorem comap_C_span_le_bot : kIdeal.comap (C : ZMod 2 →+* MvPolynomial (Fin 3) (ZMod 2)) ≤ ⊥ := by
```

Comment:
Why stating this as `≤ ⊥` rather than `= ⊥`? I mean, they're of course the same, but `=⊥` looks more natural.

Reply:
I don't remember, now fixed.